src/quick-lint-js/parse.h

```diff
@@ -3026,6 +3026,23 @@ class parser {
               });
           goto initialize_variable;
 
+        case token_type::kw_typeof:
+        case token_type::kw_class:
+        case token_type::kw_new: {
+          bool has_leading_newline = this->peek().has_leading_newline;
+          if (!has_leading_newline) {
+            const char8 *here = this->lexer_.end_of_previous_token();
+            this->error_reporter_->report(error_missing_equal_after_variable{
+                .expected_equal = source_code_span(here, here),
+            });
+          }
+          this->visit_binding_element(variable, v, declaration_kind);
+          this->skip();
+          this->parse_and_visit_expression(
+              v, precedence{.commas = false, .in_operator = allow_in_operator});
```

Comment:
Blocking: I think this breaks parsing of the following code:
```js
let x
class C {}
x = new C();
```
Please add a test and fix this. I think the problem is that you're parsing and visiting an expression even if `has_leading_newline` is true. If `has_leading_newline` is true, we should automatically insert a semicolon and end variable declaration statement parsing.